tests/test_geoserver.py

```diff
@@ -46,17 +60,89 @@ def get_geoserver_settings():
     return geoserver_settings
 
 
+def prepare_geoserver_test_workspace(test_instance, geoserver_handler, workspace_key):
+    """
+    Prepares a workspace, its datastore and a test shapefile along with the associated Geoserver resources.
+    """
+    # Creating workspace and datastore needed to publish a shapefile
+    workspace_name = test_instance.workspaces[workspace_key]
+    datastore_name = f"test-datastore-{workspace_key}"
+    geoserver_workspace_path = f"/user_workspaces/{workspace_name}/shapefile_datastore"
+    geoserver_handler._create_workspace_request(workspace_name=workspace_name)
+    geoserver_handler._create_datastore_request(workspace_name=workspace_name,
+                                                datastore_name=datastore_name)
+    geoserver_handler._configure_datastore_request(workspace_name=workspace_name,
+                                                   datastore_name=datastore_name,
+                                                   datastore_path=geoserver_workspace_path)
+
+    # Preparations needed to make tests work without all the other handlers running
+    datastore_path = get_datastore_path(test_instance.workspace_folders[workspace_key])
+
+    # This next part can fail if the user running this test doesn't have write access to the directory
+    copy_shapefile(basename=test_instance.test_shapefile_name, destination=datastore_path)
+
+    return workspace_name, datastore_name
+
+
+def reset_geoserver_test_workspace(test_instance, geoserver_handler):
+    """
+    Removes a workspace on the file system and the associated resources on Geoserver.
+    """
+    for _, workspace in test_instance.workspaces.items():
+        try:
+            geoserver_handler._remove_workspace_request(workspace_name=workspace)
+        except GeoserverError:
+            # Making sure all test workspaces are removed
+            pass
+    for _, folder in test_instance.workspace_folders.items():
+        try:
+            # Make sure access permissions are enabled before deleting files
+            os.chmod(folder, 0o777)
```

Comment:
Can we do a few checks before this, similar to below `folder == "/"`, to make sure we don't cause some troublesome modifications by mistake? Most probably they should always be located under some kind of temp directory. Abort the operation if some temp dir pattern is not detected.